pyproject.toml

```diff
@@ -40,6 +40,7 @@ cdktf = "^0.20.5"
 click = "8.1.7"
 docker = "7.1.0"
 httpx = ">=0.21.2"
+openai = "1.58.1"
```

Comment:
Does this mean we'll have a specific version of the OpenAI SDK as a dependency? That doesn't feel great since it might conflict with customer versions. Can we at least mark this as an extra?

Reply:
I'm not too worried about sub-dep conflict here since in theory they are using vellum and not the openai sdk.

however, the way around this is to specify a range like we do with the `httpx` dep above. So pick the minimum version that is supported by the code (eg let's say it's 1.0.0) and declare `openai = ">= 1.0.0"`